static/src/javascripts/projects/common/modules/commercial/acquisitions-link-enrichment.js

```diff
@@ -0,0 +1,51 @@
+// @flow
+import { addReferrerData } from 'common/modules/commercial/acquisitions-ophan';
+
+const addReferrerDataToAcquisitionLink = (rawUrl: string): string => {
+    const acquisitionDataField = 'acquisitionData';
+
+    let url;
+    try {
+        url = new URL(rawUrl);
+    } catch (e) {
+        return rawUrl;
+    }
+
+    let acquisitionData;
+    try {
+        acquisitionData = JSON.parse(
+            url.searchParams.get(acquisitionDataField)
+        );
+    } catch (e) {
+        return rawUrl;
+    }
+
+    if (acquisitionData) {
+        acquisitionData = addReferrerData(acquisitionData);
+        url.searchParams.set(
+            acquisitionDataField,
+            JSON.stringify(acquisitionData)
+        );
+    }
+
+    return url.toString();
+};
+
+const ACQUISITION_LINK_CLASS = 'js-acquisition-link';
+
+const addReferrerDataToAcquisitionLinksOnPage = (): void => {
+    const links = Array.from(
+        document.getElementsByClassName(ACQUISITION_LINK_CLASS)
+    );
+
+    links.forEach(el => {
+        const link = el.getAttribute('href');
+        if (link) {
+            el.setAttribute('href', addReferrerDataToAcquisitionLink(link));
+        }
```

Comment:
Can we wrap this using `fastdom`?
```js
fastom.read(() => el.getEttribute('href'))
  .then(link => {
    if (link) {
      fastdom.write(() => {
        el.setAttribute('href', addReferrerDataToAcquisitionLink(link));
      })
    }
  );
```

Reply:
Oh yes! I remember you saying something about this. Doing it now, thanks for the advice.